central/reports/validation/validate.go

```diff
@@ -2,34 +2,189 @@ package validation
+		if notifier.GetEmailConfig() == nil {
+			return errors.Wrap(errox.InvalidArgs, "Notifier must specify an email notifier configuration")
+		}
+		if err := v.validateEmailConfig(notifier.GetEmailConfig()); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (v *Validator) validateEmailConfig(emailConfig *apiV2.EmailNotifierConfiguration) error {
+	if emailConfig.GetNotifierId() == "" {
+		return errors.Wrap(errox.InvalidArgs, "Report configuration must specify a valid email notifier")
+	}
+	if len(emailConfig.GetMailingLists()) == 0 {
+		return errors.Wrap(errox.InvalidArgs, "Report configuration must specify at least one email recipient to send the report to")
+	}
+
+	errorList := errorhelpers.NewErrorList("Invalid email addresses in mailing list: ")
+	for _, addr := range emailConfig.GetMailingLists() {
+		if _, err := mail.ParseAddress(addr); err != nil {
+			errorList.AddError(errors.Wrapf(errox.InvalidArgs, "Invalid email recipient address: %s", addr))
+		}
+	}
+	if !errorList.Empty() {
+		return errorList.ToError()
+	}
+
+	// Use allAccessCtx since report creator/updater might not have permissions for integrationSAC
+	exists, err := v.notifierDatastore.Exists(allAccessCtx, emailConfig.GetNotifierId())
+	if err != nil {
+		return errors.Errorf("Error looking up attached notifier, Notifier ID: %s, Error: %s", emailConfig.GetNotifierId(), err)
+	}
+	if !exists {
+		return errors.Wrapf(errox.NotFound, "Notifier with ID %s not found.", emailConfig.GetNotifierId())
+	}
+	return nil
+}
+
+func (v *Validator) validateResourceScope(config *apiV2.ReportConfiguration) error {
+	if config.GetResourceScope() == nil || config.GetResourceScope().GetCollectionScope() == nil {
+		return errors.Wrap(errox.InvalidArgs, "Report configuration must specify a valid resource scope")
+	}
+	collectionID := config.GetResourceScope().GetCollectionScope().GetCollectionId()
+
+	if collectionID == "" {
+		return errors.Wrap(errox.InvalidArgs, "Report configuration must specify a valid collection ID")
+	}
+
+	// Use allAccessCtx since report creator/updater might not have permissions for workflowAdministrationSAC
+	exists, err := v.collectionDatastore.Exists(allAccessCtx, collectionID)
+	if err != nil {
+		return errors.Errorf("Error trying to lookup attached collection, Collection: %s, Error: %s", collectionID, err)
+	}
+	if !exists {
+		return errors.Wrapf(errox.NotFound, "Collection %s not found.", collectionID)
+	}
+
+	return nil
+}
+
+func (v *Validator) validateReportFilters(config *apiV2.ReportConfiguration) error {
+	if config.GetVulnReportFilters() == nil {
+		return errors.Wrap(errox.InvalidArgs, "Report configuration must include Vulnerability report filters")
+	}
+	if config.GetVulnReportFilters().GetCvesSince() == nil {
+		return errors.Wrap(errox.InvalidArgs, "Vulnerability report filters must specify how far back in time to look for CVEs. "+
+			"The valid options are 'since last successful report', 'all CVEs', and 'since a custom timestamp'")
```

Comment:
From an API standpoint, the aforementioned values do not exist, however, adding quotes indicates that the quoted fields exist.

Reply:
Yeah these were the options as shown in UI. I changed it to values that are in the API.